### PR TITLE
refac: Bundle retention

### PR DESCRIPTION
**Pull request by lasrinnil**

### CODE REVIEW

File: source/BuildingBlocks.Infrastructure/FileStorage/DataLakeFileStorageClient.cs [hunk 1]
```diff
@@ -76,4 +76,13 @@ public async Task<FileStorageFile> DownloadAsync(FileStorageReference reference)
...
     }
+
+    public async Task DeleteIfExistsAsync(FileStorageReference reference)
+    {
+        var container = _blobServiceClient.GetBlobContainerClient(reference.Category.Value);
+
+        var blob = container.GetBlobClient(reference.Path);
+
+        await blob.DeleteAsync(DeleteSnapshotsOption.IncludeSnapshots).ConfigureAwait(false);
+    }
```
Comment (ebbeknudsen) [inline source/BuildingBlocks.Infrastructure/FileStorage/DataLakeFileStorageClient.cs:87]: Where is the "if exists" part? Wouldn't this throw an exception if the file storage reference doesn't exist

File: source/IntegrationTests/Factories/WholesaleAmountPerChargeDtoBuilder.cs [hunk 2]
```diff
@@ -63,9 +63,15 @@ public WholesaleAmountPerChargeDtoBuilder WithCalculationResultId(Guid calculati
...
 
-    public WholesaleAmountPerChargeDtoBuilder WithReceiverNumber(string receiverIdValue)
+    public WholesaleAmountPerChargeDtoBuilder WithReceiverNumber(ActorNumber receiverIdValue)
     {
-        _receiverNumber = ActorNumber.Create(receiverIdValue);
+        _receiverNumber = receiverIdValue;
+        return this;
+    }
+
+    public WholesaleAmountPerChargeDtoBuilder WithChargeOwnerNumber(ActorNumber receiverIdValue)
```
Comment (ebbeknudsen) [inline source/IntegrationTests/Factories/WholesaleAmountPerChargeDtoBuilder.cs:72]:
```suggestion
    public WholesaleAmountPerChargeDtoBuilder WithChargeOwnerNumber(ActorNumber chargeOwner)
```

Comment (ebbeknudsen) [inline source/IntegrationTests/Factories/WholesaleAmountPerChargeDtoBuilder.cs:72]:
```suggestion
    public WholesaleAmountPerChargeDtoBuilder WithChargeOwnerNumber(ActorNumber chargeOwner)
```

File: source/IntegrationTests/Factories/WholesaleAmountPerChargeDtoBuilder.cs [hunk 3]
```diff
@@ -63,9 +63,15 @@ public WholesaleAmountPerChargeDtoBuilder WithCalculationResultId(Guid calculati
...
+    {
+        _chargeOwnerId = receiverIdValue;
```
Comment (ebbeknudsen) [inline source/IntegrationTests/Factories/WholesaleAmountPerChargeDtoBuilder.cs:74]:
```suggestion
        _chargeOwnerId = chargeOwner;
```

File: source/OutgoingMessages.Domain/Models/MarketDocuments/IMarketDocumentRepository.cs [hunk 4]
```diff
@@ -31,4 +30,9 @@ public interface IMarketDocumentRepository
     /// Add document to repository
     /// </summary>
     void Add(MarketDocument marketDocument);
+
+    /// <summary>
+    /// Delete market document if it exists
+    /// </summary>
+    Task DeleteMarketIfExistsDocumentAsync(BundleId bundleMessageId);
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Domain/Models/MarketDocuments/IMarketDocumentRepository.cs:37]:
```suggestion
    Task DeleteMarketDocumentIfExistsAsync(BundleId bundleMessageId);
``` 😄

File: source/OutgoingMessages.Infrastructure/DequeuedBundlesRetention.cs [hunk 5]
```diff
@@ -12,58 +12,81 @@
...
-            await command.ExecuteNonQueryAsync(cancellationToken).ConfigureAwait(false);
-            await transaction.CommitAsync(cancellationToken).ConfigureAwait(false);
-        }
-        catch (DbException)
-        {
-            // Add exception logging
-            await transaction.RollbackAsync(cancellationToken).ConfigureAwait(false);
-            throw; // re-throw exception
+            skip += incrementer;
+            take += incrementer;
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Infrastructure/DequeuedBundlesRetention.cs:89]: Shouldn't take count be a constant? So we always take, lets say 10 messages at a time

File: source/OutgoingMessages.Infrastructure/DequeuedBundlesRetention.cs [hunk 6]
```diff
@@ -12,58 +12,81 @@
...
-            FROM [dbo].[Bundles]
-            WHERE [IsDequeued] = 1)
-
-            DELETE FROM [dbo].[OutgoingMessages]
-            WHERE [AssignedBundleId] IN (SELECT [Id]
-            FROM [dbo].[Bundles]
-            WHERE [IsDequeued] = 1)
+        const int incrementer = 10;
+        var skip = 0;
+        var take = 10;
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Infrastructure/DequeuedBundlesRetention.cs:54]: I don't think you need both `incrementer` and `take`. Don't they both describe how many messages you want to "take" in each loop? (which would always be 10)

File: source/OutgoingMessages.Infrastructure/Repositories/ActorMessageQueues/ActorMessageQueueRepository.cs [hunk 7]
```diff
@@ -48,6 +49,11 @@ public ActorMessageQueueRepository(ActorMessageQueueContext actorMessageQueueCon
         return actorMessageQueue;
     }
 
+    public async Task<ReadOnlyCollection<ActorMessageQueue>> GetActorMessageQueuesAsync(int skip, int take)
+    {
+        return new ReadOnlyCollection<ActorMessageQueue>(await _actorMessageQueueContext.ActorMessageQueues.Skip(skip).Take(take).ToListAsync().ConfigureAwait(false));
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Infrastructure/Repositories/ActorMessageQueues/ActorMessageQueueRepository.cs:54]: If you use `IReadOnlyCollection` as the return value you could just return the list from the method (you wouldn't need to `new ReadOnlyCollection(<list>)`)

File: source/OutgoingMessages.Infrastructure/Repositories/OutgoingMessages/OutgoingMessageRepository.cs [hunk 8]
```diff
@@ -80,6 +81,16 @@ public async Task<OutgoingMessageBundle> GetAsync(PeekResult peekResult)
                                                                         x.ExternalId == externalId).ConfigureAwait(false);
     }
 
+    public async Task DeleteOutgoingMessageIfExistsAsync(BundleId bundleMessageId)
+    {
+        var outgoingMessage = await _context.OutgoingMessages.FirstOrDefaultAsync(x => x.AssignedBundleId == bundleMessageId).ConfigureAwait(false);
+        if (outgoingMessage is not null)
+        {
+            _context.Remove(outgoingMessage);
+            await _fileStorageClient.DeleteIfExistsAsync(outgoingMessage.FileStorageReference).ConfigureAwait(false);
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Infrastructure/Repositories/OutgoingMessages/OutgoingMessageRepository.cs:90]: Maybe we should delete on filestorageclient before removing from the context, to ensure that the outgoing message will never be removed from the context if removing from the file storage fails

File: source/OutgoingMessages.Domain/Models/MarketDocuments/IMarketDocumentRepository.cs (hunk 4 above)
Comment (lasrinnil) [reply source/OutgoingMessages.Domain/Models/MarketDocuments/IMarketDocumentRepository.cs:37]: 😆

File: source/BuildingBlocks.Infrastructure/FileStorage/DataLakeFileStorageClient.cs (hunk 1 above)
Comment (lasrinnil) [reply source/BuildingBlocks.Infrastructure/FileStorage/DataLakeFileStorageClient.cs:87]: Wups thanks, fixed it :)

**Review by ebbeknudsen — APPROVED** [review]
LGTM

**Pull request merged**